wordpress-mobile/WordPress-Android
# Issue/social 2fa sms

### Fix
Allow a code to be sent via SMS by tapping the ***Text me a code instead.*** button on the two-factor authentication screen during social login.

### Test
To focus only on the changes in this pull request, [connect a Google account to a WordPress account](https://wordpress.com/me/security/social-login) and [enable two-factor authentication](https://wordpress.com/me/security/two-step) prior to performing the steps below.

0. Clear app data.
1. Tap ***Log In*** button.
2. Tap Google login view.
3. Choose Google account.
4. Notice ***Two-Factor Authentication*** screen is shown.
5. Tap ***Text me a code instead.*** button.
6. Notice label and button text have changed.
7. Notice code is sent via SMS to phone number for account.
8. Enter code from SMS into ***Verification code*** field.
9. Tap ***Next*** button.
10. Notice ***Epilogue*** screen is shown.

#### Note
For the copy review, "Text message sent to phone number ending in %s. Please enter the verification code from the SMS." and "Text me another code instead." were added to replace "Almost there! Please enter the verification code from your Authenticator app." and "Text me a code instead." on the two-factor authentication screen after a code is sent via SMS.  See the screenshots below for illustration.

![sms](https://user-images.githubusercontent.com/3827611/32509234-f960e1fe-c3a9-11e7-9421-3c6f0365fe7c.png)
